akka-actor/src/main/scala-2.12/akka/util/ByteString.scala

```diff
@@ -417,6 +419,11 @@ object ByteString {
     }
 
     protected def writeReplace(): AnyRef = new SerializationProxy(this)
+
+    override def toArrayUnsafe(): Array[Byte] = {
```

Comment:
@johanandren Doesn't this sort of violate the expectation if it is allowed to allocate anyway?

Reply:
It must allocate an array somehow to be able to support rope-ByteString. The intention of this was to use it for the ByteString1 case. We can skip it and use `asReadOnlyByteBuffer` for that instead. One more allocation, but that's ok compared to all other things.